service/src/test/java/uk/gov/hmcts/reform/fpl/service/docmosis/DischargeCareOrderGenerationServiceTest.java

```diff
@@ -0,0 +1,182 @@
+import uk.gov.hmcts.reform.fpl.service.HearingVenueLookUpService;
+import uk.gov.hmcts.reform.fpl.service.config.LookupTestConfig;
+import uk.gov.hmcts.reform.fpl.utils.FixedTimeConfiguration;
+
+import java.time.LocalDate;
+import java.time.format.FormatStyle;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Optional.ofNullable;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.enums.ChildGender.BOY;
+import static uk.gov.hmcts.reform.fpl.enums.ChildGender.GIRL;
+import static uk.gov.hmcts.reform.fpl.enums.GeneratedOrderType.DISCHARGE_OF_CARE_ORDER;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateToString;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testChild;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocumentReference;
+
+@SuppressWarnings("unchecked")
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {DischargeCareOrderGenerationService.class, CaseDataExtractionService.class,
+    LookupTestConfig.class, HearingVenueLookUpService.class, JacksonAutoConfiguration.class,
+    FixedTimeConfiguration.class, ChildrenService.class, DischargeCareOrderService.class})
+class DischargeCareOrderGenerationServiceTest extends AbstractOrderGenerationServiceTest {
+
+    @Autowired
+    private DischargeCareOrderGenerationService service;
+
+    private GeneratedOrder orderWithMultipleChildren = GeneratedOrder.builder()
+        .dateOfIssue("1 Jun 2020")
+        .children(List.of(
+            testChild("John", "Smith", BOY, LocalDate.of(2018, 1, 1)),
+            testChild("Eva", "Smith", GIRL, LocalDate.of(2016, 1, 1))
+        ))
+        .type("Interim care order")
+        .courtName("Newcastle Court")
+        .build();
+
+    private GeneratedOrder orderWithSingleChild = GeneratedOrder.builder()
+        .dateOfIssue("2 Jun 2020")
+        .children(List.of(
+            testChild("George", "West", BOY, LocalDate.of(2019, 1, 1))
+        ))
+        .courtName("Reading Court")
+        .type("Final care order")
+        .build();
+
+    private GeneratedOrder legacyOrder = GeneratedOrder.builder()
+        .dateOfIssue("2 Jun 2020")
+        .type("Final care order")
+        .build();
+
+    @ParameterizedTest
+    @EnumSource(OrderStatus.class)
+    void shouldGenerateTemplateDataWhenOneOutOfMultipleCareOrdersSelected(OrderStatus orderStatus) {
+        List<Integer> selectedOrders = List.of(0);
+        CaseData caseData = getCase(orderStatus, selectedOrders, orderWithMultipleChildren, orderWithSingleChild);
+
+        DocmosisGeneratedOrder actualOrderData = service.getTemplateData(caseData);
+
+        String expectedDetails = "The court discharges the care order made by the Newcastle Court on 1 Jun 2020";
```

Comment:
Hmm double checking, are we happy with having short month name in generated docs? I would say in other orders we use the full form. Can you pllease check?

Reply:
It is because of test data of this test, in reality it is long name, will update test data to not make confusion